File: DynamicKey/AgoraDynamicKey/lua-resty/sample/rtc_token_builder_sample.lua
local ngx = ngx
local require = require
local rtc_token_builder = require "resty.agoradynamickey.rtc_token_builder"
local now = ngx.now

local app_id = "970CA35de60c44645bbae8a215061b33"
local app_certificate = "5CFd2fd1755d40ecb72977518be15d3b"
local channel_name = "7d72365eb983485397e3e3f9d460bdda"
local uid = 2882341273
local user_account = "2882341273"
local expire_delay_ts = 3600
local current_ts = now()
local privilege_expired_ts = current_ts + expire_delay_ts
local role = rtc_token_builder.Role_Attendee

local token = rtc_token_builder.build_token_with_uid(app_id, app_certificate, channel_name, uid, role, privilege_expired_ts)
ngx.log(ngx.ERR, "Token with int uid: ", token)
token = rtc_token_builder.build_token_with_account(app_id, app_certificate, channel_name, user_account, role, privilege_expired_ts)
ngx.log(ngx.ERR, "Token with user account: ", token)

File: DynamicKey/AgoraDynamicKey/nodejs/sample/RtcTokenBuilderSample.js
const RtcTokenBuilder = require('../src/RtcTokenBuilder').RtcTokenBuilder;
const RtcRole = require('../src/RtcTokenBuilder').Role;

const appID = '1ff2a19999bd457b82e823d5b69d3465';
const appCertificate = '2246825227f64a21abd056239ed77bed';
const channelName = '1081460522';
const uid = 2882341273;
const account = "2882341273";
const role = RtcRole.PUBLISHER;

const expirationTimeInSeconds = 3600

const currentTimestamp = Math.floor(Date.now() / 1000)

const privilegeExpiredTs = currentTimestamp + expirationTimeInSeconds

// IMPORTANT! Build token with either the uid or with the user account. Comment out the option you do not want to use below.

// Build token with uid
const tokenA = RtcTokenBuilder.buildTokenWithUid(appID, appCertificate, channelName, uid, role, privilegeExpiredTs);
console.log("Token With Integer Number Uid: " + tokenA);

// Build token with user account
const tokenB = RtcTokenBuilder.buildTokenWithAccount(appID, appCertificate, channelName, account, role, privilegeExpiredTs);
console.log("Token With UserAccount: " + tokenB);
